codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -316,10 +332,10 @@ public Http2Stream setPriority(int parentStreamId, short weight, boolean exclusi
             // Already have a priority. Re-prioritize the stream.
             weight(weight);
 
-            if (newParent != parent() || exclusive) {
+            if (newParent != parent() || (exclusive && newParent.numChildren() != 1)) {
```

Comment:
@Scottmitch Oh I see ... by the time we're checking exclusive we know that the parent isn't changing, we're handling the exclusive case.  Yeah this change looks right to me.